lib/mongoid/association/depending.rb

```diff
@@ -10,7 +10,15 @@ module Depending
 
       included do
         class_attribute :dependents
+        class_attribute :dependents_owner
         self.dependents = []
+        self.dependents_owner = self
+      end
+
+      class_methods do
+        def _all_dependents
+          dependents + (superclass.respond_to?(:_all_dependents) ? superclass._all_dependents : [])
```

Comment:
This needs to deduplicate the list by association name.

Reply:
Oh, good catch!